File: job-man-core/src/main/scala/pl/newicom/jobman/DistributedPubSubFacade.scala
package pl.newicom.jobman

import akka.actor.ActorRef
import akka.actor.typed.scaladsl.adapter._
import akka.cluster.pubsub.DistributedPubSubMediator

class DistributedPubSubFacade(mediator: ActorRef) {

  def subscribe(topic: String, ref: akka.actor.typed.ActorRef[_]): Unit =
    subscribe(topic, ref.toUntyped)

  def subscribe(topic: String, ref: ActorRef): Unit =
    mediator ! new DistributedPubSubMediator.Subscribe(topic, ref)

  def publish(topic: String, msg: Any): Unit =
    mediator ! DistributedPubSubMediator.Publish(topic, msg)

}

File: job-man-view-sql/src/main/java/pl/newicom/jobman/view/sql/DistributedPubSubFacade.java
package pl.newicom.jobman.view.sql;

import static akka.actor.ActorRef.noSender;
import static akka.actor.typed.javadsl.Adapter.toUntyped;

import akka.actor.ActorRef;
import akka.cluster.pubsub.DistributedPubSubMediator;

public class DistributedPubSubFacade {
	private final ActorRef mediator;

	DistributedPubSubFacade(ActorRef mediator) {
		this.mediator = mediator;
	}

	public void subscribe(String topic, akka.actor.typed.ActorRef<?> ref) {
		subscribe(topic, toUntyped(ref));
	}

	public void subscribe(String topic, ActorRef ref) {
		mediator.tell(new DistributedPubSubMediator.Subscribe(topic, ref), noSender());
	}

	public void publish(String topic, Object msg) {
		mediator.tell(new DistributedPubSubMediator.Publish(topic, msg), noSender());
	}
}
